gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableDataset.java

```diff
@@ -25,7 +25,7 @@
 /**
  * {@link Dataset} that supports finding {@link CopyableFile}s.
  */
-public interface CopyableDataset extends PartitionableDataset<CopyableFile> {
+public interface CopyableDataset extends Dataset {
 
   /**
    * Find all {@link CopyableFile}s in this dataset.
```

Comment:
The javadoc needs to be made more precise? Can the result contain directories? Shouldn't the return result be a Collection ? I can't think of a case where the order is important.